src/Castle.Facilities.Logging/LoggingFacility.cs

```diff
@@ -12,403 +12,43 @@
-		/// </summary>
-		public LoggingFacility()
-		{
-		}
-
-		/// <summary>
-		///   Initializes a new instance of the <see cref="LoggingFacility" /> class.
-		/// </summary>
-		/// <param name="loggingApi"> The LoggerImplementation that should be used </param>
-		[Obsolete("A logger factory implementation type should be provided via LogUsing<T>(), this will be removed in the future.")]
-		public LoggingFacility(LoggerImplementation loggingApi) : this(loggingApi, null)
-		{
-		}
-
-		/// <summary>
-		///   Initializes a new instance of the <see cref="LoggingFacility" /> class.
-		/// </summary>
-		/// <param name="loggingApi"> The LoggerImplementation that should be used </param>
-		/// <param name="configFile"> The configuration file that should be used by the chosen LoggerImplementation </param>
-		[Obsolete("A logger factory implementation type should be provided via LogUsing<T>(), this will be removed in the future.")]
-		public LoggingFacility(LoggerImplementation loggingApi, string configFile) : this(loggingApi, null, configFile)
-		{
-		}
-
-		/// <summary>
-		///   Initializes a new instance of the <see cref="LoggingFacility" /> class using a custom LoggerImplementation
-		/// </summary>
-		/// <param name="configFile"> The configuration file that should be used by the chosen LoggerImplementation </param>
-		/// <param name="customLoggerFactory"> The type name of the type of the custom logger factory. </param>
-		public LoggingFacility(string customLoggerFactory, string configFile)
-			: this(LoggerImplementation.Custom, customLoggerFactory, configFile)
-		{
-		}
-
-		/// <summary>
-		///   Initializes a new instance of the <see cref="LoggingFacility" /> class.
-		/// </summary>
-		/// <param name="loggingApi"> The LoggerImplementation that should be used </param>
-		/// <param name="configFile"> The configuration file that should be used by the chosen LoggerImplementation </param>
-		/// <param name="customLoggerFactory"> The type name of the type of the custom logger factory. (only used when loggingApi is set to LoggerImplementation.Custom) </param>
-		[Obsolete("A logger factory implementation type should be provided via LogUsing<T>(), this will be removed in the future.")]
-		public LoggingFacility(LoggerImplementation loggingApi, string customLoggerFactory, string configFile)
-		{
-			loggerImplementation = loggingApi;
-			customLoggerFactoryTypeName = customLoggerFactory;
-			configFileName = configFile;
-		}
-
-		[Obsolete("A logger factory implementation type should be provided via LogUsing<T>(), this will be removed in the future.")]
-		public LoggingFacility LogUsing(LoggerImplementation loggingApi)
-		{
-			if (loggingApi == LoggerImplementation.Custom)
-			{
-				throw new FacilityException("To use custom logger use LogUsing<TLoggerFactory>() method.");
-			}
-			loggerImplementation = loggingApi;
-			return this;
-		}
-
-		public LoggingFacility LogUsing<TLoggerFactory>()
```

Comment:
Removing this obviously means the logger factory can't automatically have dependencies resolved from the container. If this is the direction we want to take deprecating this API now would be the way to go, otherwise our documentation literally tells people to do something that we are removing.

Reply:
This was perhaps and eager cut, just that I saw the richness of the constructors for logger factories and thought this potentially hides features that are discoverable. Can bring this back but perhaps will refer back to the issue for further discussion before we accept this.